src/Symfony/Component/Serializer/Serializer.php

```diff
@@ -127,10 +127,8 @@ public function normalize($data, $format = null, array $context = array())
         if (null === $data || is_scalar($data)) {
             return $data;
         }
-        if (is_object($data) && $this->supportsNormalization($data, $format)) {
-            return $this->normalizeObject($data, $format, $context);
-        }
-        if ($data instanceof \Traversable) {
+
+        if (is_array($data) || $data instanceof \Traversable) {
```

Comment:
@dunglas Custom normalizers are called first at the top of the method (normalizers were iterated 3 times in this method before the refactoring in this PR).